src/input/datetime.rs

```diff
@@ -563,8 +563,16 @@ impl TzInfo {
         hasher.finish()
     }
 
-    fn __richcmp__(&self, other: &Self, op: CompareOp) -> bool {
-        op.matches(self.seconds.cmp(&other.seconds))
+    fn __richcmp__(&self, other: &PyAny, op: CompareOp) -> PyResult<Py<PyAny>> {
+        let py = other.py();
+        if other.is_instance_of::<PyTzInfo>() {
+            let offset_delta = other.call_method1(intern!(py, "utcoffset"), (py.None().as_ref(py),))?;
```

Comment:
Should work without the `.as_ref(py)` here.
```suggestion
            let offset_delta = other.call_method1(intern!(py, "utcoffset"), (py.None(),))?;
```